llvm/tools/sycl-post-link/SYCLDeviceRequirements.cpp

```diff
@@ -16,26 +16,30 @@
 
 using namespace llvm;
 
-std::map<StringRef, std::vector<uint32_t>>
-llvm::getSYCLDeviceRequirements(const Module &M) {
-  std::map<StringRef, std::vector<uint32_t>> Result;
+void llvm::getSYCLDeviceRequirements(
+    const Module &M, std::map<StringRef, std::vector<uint32_t>> &Requirements) {
   auto ExtractIntegerFromMDNodeOperand = [=](const MDNode *N,
                                              unsigned OpNo) -> unsigned {
     Constant *C =
         cast<ConstantAsMetadata>(N->getOperand(OpNo).get())->getValue();
     return static_cast<uint32_t>(C->getUniqueInteger().getZExtValue());
   };
-  std::set<uint32_t> Aspects;
-  for (const Function &F : M) {
-    if (!F.hasMetadata("sycl_used_aspects"))
-      continue;
 
-    const MDNode *MD = F.getMetadata("sycl_used_aspects");
-    for (size_t I = 0, E = MD->getNumOperands(); I < E; ++I) {
-      Aspects.insert(ExtractIntegerFromMDNodeOperand(MD, I));
+  constexpr std::pair<const char *, const char *> ReqdMDs[] = {
```

Comment:
Can we please have a comment here stating a reference for where these strings come from? Thanks